catalog/plugin/catalog-plugin-querymonitor/src/test/java/ddf/catalog/querymonitor/QueryMonitorTest.java

```diff
@@ -0,0 +1,243 @@
+        CatalogProvider mockSource = mock(CatalogProvider.class);
+        uuid = UUID.randomUUID();
+        as = new ActiveSearch(null, mockSource, uuid, CLIENT_TEXT);
+        assertThat(as.getSource(), is(mockSource));
+        assertThat(as.getUniqueID(), is(uuid));
+        assertThat(as.getClientInfo(), is(CLIENT_TEXT));
+    }
+
+    @Test
+    public void testActiveSearchConstructorValidInputs() {
+        CatalogProvider mockSource = mock(CatalogProvider.class);
+        uuid = UUID.randomUUID();
+        as = new ActiveSearch(CQL_TEXT, mockSource, uuid, CLIENT_TEXT);
+        assertThat(as.getCQL(), is(CQL_TEXT));
+        assertThat(as.getSource(), is(mockSource));
+        assertThat(as.getUniqueID(), is(uuid));
+        assertThat(as.getClientInfo(), is(CLIENT_TEXT));
+    }
+
+    @Test
+    public void testQueryMonitorAddActiveSearchParameterNull() {
+        as = null;
+        assertThat(qmpi.addActiveSearch(as), is(false));
+    }
+
+    @Test
+    public void testQueryMonitorAddActiveSearchWithValidInput() {
+        as = new ActiveSearch(null, null, null, null);
+        assertThat(qmpi.addActiveSearch(as), is(true));
+    }
+
+    @Test
+    public void testQueryMonitorPluginImplRemoveActiveSearchUsingUUIDNullUUID()
+            throws StopProcessingException, PluginExecutionException {
+        QueryRequest mockQR = mock(QueryRequest.class);
+        propertyMap = new ConcurrentHashMap<>();
+        when(mockQR.getProperties()).thenReturn(propertyMap);
+        when(mockQR.toString()).thenReturn(QUERY_REQUEST_TEXT);
+        qmpi.setRemoveSearchAfterComplete(true);
+        qmpi.process(null, mockQR);
+        Map<UUID, ActiveSearch> activeSearchTable = qmpi.getActiveSearches();
+        if (activeSearchTable != null && !activeSearchTable.isEmpty()) {
+            Iterator it = activeSearchTable.entrySet()
+                    .iterator();
+            if (it != null && it.hasNext()) {
+                Map.Entry<UUID, ActiveSearch> tempEntry = (Map.Entry) it.next();
+                as = tempEntry.getValue();
+                UUID u = null;
+                assertThat(qmpi.removeActiveSearch(u), is(false));
+                u = as.getUniqueID();
+                qmpi.removeActiveSearch(u);
+                return;
+            }
+        }
+        Assert.fail();
+    }
+
+    @Test
+    public void testQueryMonitorPluginImplRemoveActiveSearchUsingUUIDValidUUID()
+            throws StopProcessingException, PluginExecutionException {
+        QueryRequest mockQR = mock(QueryRequest.class);
+        propertyMap = new ConcurrentHashMap<>();
+        when(mockQR.getProperties()).thenReturn(propertyMap);
+        when(mockQR.toString()).thenReturn(QUERY_REQUEST_TEXT);
+        qmpi.setRemoveSearchAfterComplete(true);
+        qmpi.process(null, mockQR);
+        Map<UUID, ActiveSearch> activeSearchTable = qmpi.getActiveSearches();
+        if (activeSearchTable != null && !activeSearchTable.isEmpty()) {
```

Comment:
We are upgrading hamcrest to version 1.3. I found that avoiding **is** is good in v1.3.

It's hard to be sure of the particulars, but it looks like the test is to ensure that a search is removed after it completes. That said, I'm not sure that `qmpi.removeActiveSearch(null)` is a useful test. I think lines 129-143 could be replaced with:

```
UUID uuid  = mpi.getActiveSearches().values().get(0).getUniqueID();
assertThat(qmpi.removeActiveSearch(uuid));
````
If something is null or null or empty along the way, the code will throw an exception and the test will fail.